native/libcst/src/parser/errors.rs

```diff
@@ -59,36 +52,21 @@ mod py_error {
                     line = lines.len() - 1;
                     col = 0;
                 }
-                let kwargs = [
-                    ("message", e.to_string().into_py(py)),
-                    ("lines", lines.into_py(py)),
-                    ("raw_line", (line + 1).into_py(py)),
-                    ("raw_column", col.into_py(py)),
-                ]
-                .into_py_dict_bound(py);
-                let libcst =
-                    PyModule::import_bound(py, "libcst").expect("libcst cannot be imported");
-                PyErr::from_value_bound(
-                    libcst
-                        .getattr("ParserSyntaxError")
-                        .expect("ParserSyntaxError not found")
-                        .call((), Some(&kwargs))
-                        .expect("failed to instantiate"),
-                )
+                match || -> PyResult<Bound<'_, PyAny>> {
+                    let kwargs = [
+                        ("message", e.to_string().into_pyobject(py)?.into_any()),
+                        ("lines", lines.into_pyobject(py)?.into_any()),
+                        ("raw_line", (line + 1).into_pyobject(py)?.into_any()),
+                        ("raw_column", col.into_pyobject(py)?.into_any()),
+                    ]
+                    .into_py_dict(py)?;
+                    let libcst = PyModule::import(py, "libcst")?;
+                    libcst.getattr("ParserSyntaxError")?.call((), Some(&kwargs))
+                }() {
+                    Ok(py_err_value) => PyErr::from_value(py_err_value),
+                    Err(e) => e,
+                }
```

Comment:
Since ultimately we want a PyErr in both the success and error cases, we can make use of a closure that returns a result and `?` to avoid `unwrap` or `expect` in the new `into_pyobject()` calls that can return errors and at the same time clean up some unnecessary pre-existing `expect` use to turn some panics into errors. This is a neat Rust trick, unfortuntely it's only really convenient in error handling code.